src/external/bincode_support.rs

```diff
@@ -0,0 +1,259 @@
+// Copyright 2013-2014 The Rust Project Developers.
+// Copyright 2018 The Uuid Project Developers.
+//
+// See the COPYRIGHT file at the top-level directory of this distribution.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use crate::{
+    convert::TryFrom,
+    fmt::{Braced, Hyphenated, Simple, Urn},
+    non_nil::NonNilUuid,
+    Uuid,
+};
+use bincode::{
+    config::{Config, IntEncoding},
+    de::{BorrowDecoder, Decoder},
+    enc::Encoder,
+    error::{DecodeError, EncodeError},
+    BorrowDecode, Decode, Encode,
+};
+use std::string::ToString;
+
+impl Encode for Uuid {
+    fn encode<E: Encoder>(&self, encoder: &mut E) -> Result<(), EncodeError> {
+        if encoder.config().int_encoding() == IntEncoding::Fixed {
+            usize::encode(&16, encoder)?;
+        } else {
+            u8::encode(&16, encoder)?;
+        }
+
+        <&[u8; 16]>::encode(&self.as_bytes(), encoder)?;
+
+        Ok(())
```

Comment:
I believe this is sufficient and delegates the int encoding implementation to bincode itself
```suggestion
        <&[u8]>::encode(&self.as_bytes().as_slice(), encoder)
```

Reply:
Leaning on the impl provided by `bincode` itself seems like a good idea. That way we should naturally respect any new options it introduces that might affect encoding that end users may want.